calculate_daily_summary.py

```diff
@@ -90,62 +101,73 @@
     ORDER BY 1;
 """
 
+Q_MD_USERS_EXPIRE = """
+    DELETE FROM daily_multi_device_users{suffix}
+    WHERE day < '{day_expiry}'::DATE;
+"""
+
 Q_GET_FIRST_UNPROCESSED_DAY = """
     SELECT (MAX(day) + '1 day'::INTERVAL) AS timestamp
-    FROM daily_multi_device_users;
+    FROM daily_multi_device_users{suffix};
 """
 
 Q_GET_FIRST_AVAILABLE_DAY = """
     SELECT MIN(timestamp)::DATE
-    FROM activity_events;
+    FROM activity_events{suffix};
 """
 
 Q_GET_LAST_AVAILABLE_DAY = """
     SELECT MAX(timestamp)::DATE
-    FROM activity_events;
+    FROM activity_events{suffix};
 """
 
 Q_VACUUM_TABLES = """
     END;
-    VACUUM FULL daily_activity_per_device;
-    VACUUM FULL daily_multi_device_users;
+    VACUUM FULL daily_activity_per_device{suffix};
+    VACUUM FULL daily_multi_device_users{suffix};
 """
 
-def summarize_events(day_from=None, day_until=None):
+def summarize_events():
     db = postgres.Postgres(DB)
     db.run("BEGIN TRANSACTION")
     try:
-        db.run(Q_DAILY_DEVICES_CREATE_TABLE)
-        db.run(Q_MD_USERS_CREATE_TABLE)
-        # By default, summarize the latest days that are not yet summarized.
-        if day_from is None:
-            day_from = db.one(Q_GET_FIRST_UNPROCESSED_DAY)
+        for suffix in TABLE_SUFFIXES:
+            db.run(Q_DAILY_DEVICES_CREATE_TABLE.format(suffix=suffix))
+            db.run(Q_MD_USERS_CREATE_TABLE.format(suffix=suffix))
+            day_expiry = db.one(Q_GET_FIRST_AVAILABLE_DAY.format(suffix=suffix))
+            # By default, summarize the latest days that are not yet summarized.
+            day_from = db.one(Q_GET_FIRST_UNPROCESSED_DAY.format(suffix=suffix))
             if day_from is None:
-                day_from = db.one(Q_GET_FIRST_AVAILABLE_DAY)
+                day_from = day_expiry
                 if day_from is None:
                     raise RuntimeError('no events in db')
-        if day_until is None:
-            day_until = db.one(Q_GET_LAST_AVAILABLE_DAY)
-        days = {
-            "day_from": day_from,
-            "day_until": day_until,
-        }
-        print "SUMMARIZING FROM", day_from, "UNTIL", day_until
-        # Update daily device activity.
-        print "UPDATING DAILY ACTIVE DEVICES SUMMARY"
-        db.run(Q_DAILY_DEVICES_CLEAR.format(**days))
-        db.run(Q_DAILY_DEVICES_SUMMARIZE.format(**days))
-        # Update multi-device-user assessments.
-        print "UPDATING MULTI-DEVICE USERS SUMMARY"
-        db.run(Q_MD_USERS_CLEAR.format(**days))
-        db.run(Q_MD_USERS_SUMMARIZE.format(**days))
+            day_until = db.one(Q_GET_LAST_AVAILABLE_DAY.format(suffix=suffix))
+            days = {
+                "day_from": day_from,
+                "day_until": day_until,
+                "suffix": suffix
+            }
+            print "SUMMARIZING FROM", day_from, "UNTIL", day_until
```

Comment:
nit: maybe worth adding "FOR SUFFIX" to these prints, like you did with the expiry one below.